Wikipedia/Code/MWKArticle.m

```diff
@@ -431,14 +431,12 @@ - (NSString*)debugDescription {
     }];
 }
 
--(NSArray<NSURL*>*)schemelessURLs:(NSArray<NSURL*>*)urls {
-    return [urls bk_map:^NSURL*(NSURL* url){
-        NSString* urlStr = url.absoluteString;
-        NSRange dividerRange = [urlStr rangeOfString:@"://"];
-        if (dividerRange.location != NSNotFound) {
-            return [NSURL URLWithString:[urlStr substringFromIndex:dividerRange.location + 1]];
+-(NSArray<NSURL*>*)schemelessURLsRejectingNilURLs:(NSArray<NSURL*>*)urls {
+    return [urls wmf_mapAndRejectNil:^NSURL*(NSURL* url){
+        if([url isKindOfClass:[NSURL class]]){
```

Comment:
Already talked about in hangout but for posterity:

It can't be guaranteed - it is only a hint to the compiler